src/main/java/org/opentripplanner/routing/api/request/framework/TimePenalty.java

```diff
@@ -31,6 +32,13 @@ protected boolean isZero(Duration value) {
     return value.isZero();
   }
 
+  /**
+   * Does this penalty actually modify a duration or would it be returned unchanged?
+   */
+  public boolean modifies() {
+    return !constant().isZero() && coefficient() != 1.0;
```

Comment:
Is this comparison reliable or should we use a delta?

Reply:
Well, if it's not exactly 1.0 then I would argue that it does modify and therefore should return true.